src/test/java/org/mockitousage/verification/rules/JUnitVerificationCollectorTest.java

```diff
@@ -0,0 +1,132 @@
+
+public class JUnitVerificationCollectorTest {
+
+    @Test
+    public void should_not_throw_any_exceptions_when_verifications_are_succesful() {
+        JUnitVerificationCollector collector = new JUnitVerificationCollector();
+
+        IMethods methods = mock(IMethods.class);
+        methods.simpleMethod();
+
+        collector.verify(methods).simpleMethod();
+        collector.collectAndReport();
+    }
+
+    @Test(expected = MockitoAssertionError.class)
+    public void should_collect_verification_failures() {
+        JUnitVerificationCollector collector = new JUnitVerificationCollector();
+
+        IMethods methods = mock(IMethods.class);
+
+        collector.verify(methods).simpleMethod();
+        collector.collectAndReport();
+    }
+
+    @Test
+    public void should_collect_multiple_verification_failures() {
+        JUnitVerificationCollector collector = new JUnitVerificationCollector();
+
+        IMethods methods = mock(IMethods.class);
+
+        collector.verify(methods).simpleMethod();
+        collector.verify(methods).byteReturningMethod();
+        try {
+            collector.collectAndReport();
+            fail();
+        } catch (MockitoAssertionError error) {
+            assertThat(error).hasMessageContaining("1. Wanted but not invoked:");
+            assertThat(error).hasMessageContaining("2. Wanted but not invoked:");
+        }
+    }
+
+    @Test
+    public void should_only_collect_failures_ignore_succesful_verifications() {
+        JUnitVerificationCollector collector = new JUnitVerificationCollector();
+
+        IMethods methods = mock(IMethods.class);
+
+        collector.verify(methods, never()).simpleMethod();
+        collector.verify(methods).byteReturningMethod();
+        try {
+            collector.collectAndReport();
+            fail();
+        } catch (MockitoAssertionError error) {
+            assertThat(error).hasMessageContaining("1. Wanted but not invoked:");
+            assertThat(error.getMessage()).doesNotContain("2.");
+        }
+    }
+
+    @Test
+    public void should_continue_collecting_after_failing_verification() {
+        JUnitVerificationCollector collector = new JUnitVerificationCollector();
+
+        IMethods methods = mock(IMethods.class);
+        methods.simpleMethod();
+
+        collector.verify(methods).byteReturningMethod();
+        collector.verify(methods).simpleMethod();
+        try {
+            collector.collectAndReport();
+            fail();
+        } catch (MockitoAssertionError error) {
+            assertThat(error).hasMessageContaining("1. Wanted but not invoked:");
+            assertThat(error.getMessage()).doesNotContain("2.");
+        }
+    }
+
+    /*
+        Dirty fix in order to only execute the inner test class when the test case below is run.
+        Otherwise the inner test class will be executed in the suite and fail as we expect.
+     */
+    private static boolean SHOULD_RUN_RULE = false;
```

Comment:
Don't do these hacks ;)
Instead play with core JUnit functionality like `JUnitCore`, there's some example in the test code base (like `org.mockitousage.bugs.MockitoRunnerBreaksWhenNoTestMethodsTest`)